src/pyodide/internal/workers.py

```diff
@@ -149,54 +157,202 @@ def json(
+    ):
         if form_data:
             if isinstance(form_data, dict):
-                self.js_form_data = js.FormData.new()
-                for item in form_data.items():
-                    self.js_form_data.append(item[0], item[1])
+                self._js_form_data = js.FormData.new()
+                for k, v in form_data.items():
+                    self._js_form_data.append(k, _py_value_to_js(v))
             else:
-                self.js_form_data = form_data
+                self._js_form_data = form_data
         else:
-            self.js_form_data = js.FormData.new()
+            self._js_form_data = js.FormData.new()
 
-    def __getitem__(self, key: str) -> list[FormDataValue]:
-        return list(self.js_form_data.getAll(key))
+    def __getitem__(self, key: str) -> FormDataValue:
+        return _js_value_to_py(self._js_form_data.get(key))
 
-    def __setitem__(self, key: str, value: list[FormDataValue]):
-        self.js_form_data.delete(key)
-        for item in value:
-            self.js_form_data.append(key, item)
+    def __setitem__(self, key: str, value: FormDataValue):
+        if isinstance(value, list):
+            raise TypeError("Expected single item in arguments to FormData.__setitem__")
+        self._js_form_data.set(key, _py_value_to_js(value))
 
-    def append(self, key: str, value: FormDataValue):
-        self.js_form_data.append(key, value)
+    def append(self, key: str, value: FormDataValue, filename: str | None = None):
+        if filename:
+            self._js_form_data.append(key, _py_value_to_js(value), filename)
+        else:
+            self._js_form_data.append(key, _py_value_to_js(value))
 
     def delete(self, key: str):
-        self.js_form_data.delete(key)
+        self._js_form_data.delete(key)
 
     def __contains__(self, key: str) -> bool:
-        return self.js_form_data.has(key)
+        return self._js_form_data.has(key)
 
     def values(self) -> Generator[FormDataValue, None, None]:
-        yield from self.js_form_data.values()
+        for val in self._js_form_data.values():
+            yield _js_value_to_py(val)
 
     def keys(self) -> Generator[str, None, None]:
-        yield from self.js_form_data.keys()
+        yield from self._js_form_data.keys()
 
     def __iter__(self):
         yield from self.keys()
 
     def items(self) -> Generator[tuple[str, FormDataValue], None, None]:
-        for item in self.js_form_data.entries():
-            yield (item[0], item[1])
+        for item in self._js_form_data.entries():
+            yield (item[0], _js_value_to_py(item[1]))
 
     def __delitem__(self, key: str):
         self.delete(key)
 
     def __len__(self):
         return len(self.keys())
+
+    def get_all(self, key: str) -> list[FormDataValue]:
+        return list(map(lambda x: _js_value_to_py(x), self._js_form_data.getAll(key)))
+
+    @property
+    def native(self) -> "js.FormData":
+        return self._js_form_data
```

Comment:
Is this a better name than `js_form_data`? I guess using `native` everywhere is most likely advantageous so that people don't need to remember the name. Though I'd prefer something with `js` in it. What about `js_object`?

Reply:
Yeah, I was going to ask if anyone has better name suggestions. `js_object` sounds good.

Reply:
And yeah, I introduced it so that it's consistent across all the objects.